src/operations/blas3/gemm_no_local_full_vec.hpp

```diff
@@ -0,0 +1,911 @@
+          typename element_t, bool is_beta_zero, int VectorSize>
+class Gemm<input_t, output_t, DoubleBuffer, NbcA, NbcB, ClSize, tile_type,
+           TransA, TransB, element_t, is_beta_zero,
+           static_cast<int>(gemm_memory_t::no_local),
+           static_cast<int>(gemm_algorithm_t::standard),
+           static_cast<int>(gemm_vectorization_t::full), VectorSize> {
+ public:
+  using value_t = element_t;
+  using index_t = typename std::make_signed<typename input_t::index_t>::type;
+  using address_t = cl::sycl::access::address_space;
+  using packetize_t = Packetize<VectorSize, value_t, index_t>;
+  static constexpr int local_memory_size = 0;
+  /*! @brief The number of rows processed by each work item */
+  static constexpr index_t item_rows = tile_type::item_rows;
+  /*! @brief The number of cols processed by each work item */
+  static constexpr index_t item_cols = tile_type::item_cols;
+  /*! @brief The number of work items in each row of work group */
+  static constexpr index_t wg_rows = tile_type::wg_rows;
+  /*! @brief The number of work items in each column of work group */
+  static constexpr index_t wg_cols = tile_type::wg_cols;
+  /*! @brief Number of rows within a work-group level tile */
+  static constexpr index_t block_rows = wg_rows * item_rows;
+  /*! @brief Number of columns within a work-group level tile */
+  static constexpr index_t block_cols = wg_cols * item_cols;
+  /*! @brief A boolean parameter represents wheather or not matrix A is
+   * transposed */
+  static constexpr bool trans_a = TransA;
+  /*! @brief A boolean parameter represents wheather or not matrix B is
+   * transposed */
+  static constexpr bool trans_b = TransB;
+
+  static_assert(wg_cols * item_cols == item_rows * wg_rows,
+                "Block level tile should be square");
+
+  static_assert(item_rows % packetize_t::packet_size == 0,
+                "Item rows must be a multiple of the vector packet size");
+  static_assert(item_cols % packetize_t::packet_size == 0,
+                "Item cols must be a multiple of the vector packet size");
+  static_assert(
+      packetize_t::template check_size<item_rows>(),
+      "If vectorization is enabled item_rows must equal the packet_size");
+  static_assert(
+      packetize_t::template check_size<item_cols>(),
+      "If vectorization is enabled item_cols must equal the packet_size");
+
+  input_t a_;
+  input_t b_;
+  output_t c_;
+  const element_t alpha_;
+  const element_t beta_;
+  index_t batch_size_;
+  SYCL_BLAS_INLINE Gemm(input_t A, input_t B, output_t C, element_t alpha,
+                        element_t beta, index_t batch_size)
+      : a_(A),
+        b_(B),
+        c_(C),
+        alpha_(alpha),
+        beta_(beta / alpha_),
+        batch_size_(batch_size) {}
+
+  /*!
+   * @brief Get the type of this NoLocalGemmFactory as a human readable string.
+   */
+  static SYCL_BLAS_INLINE std::string get_type_string() noexcept {
```

Comment:
I don't really know where this is used, but it doesn't look like it contains any information about the vector type/size. Is that important?

Reply:
It looks like it is only used in the auto tuner which from what I gather appends the configuration information anyway.

Reply:
I've updated this to fix some inconsistencies and add the vectorize type and size. Other GEMM kernels also need this but I don't want to clutter this PR with those changes, and it's not a huge priority I think.

Reply:
Yeah if it's only used in the tuner then it's not particularly important